### DISCUSSION TITLE
piecewise linear with Yalmip

### DISCUSSION THREAD

**Discussion by BaiYansong-bys** [yalmip/YALMIP#1133] [Optimization modelling] [answered]
Hi, Johan
If I have an optimization model to linear a function by piecewise method, the objective function is as follows,
min max (a**x**+b-f(**x**))^2,(f(x) is the nonlinear function which needs to be linearized, **x** is  variable which is a vector)
the constraints are as follows,
**x_low**<=**x**<=**x_upp**
I found some papers tell that the nonlinear model can be solved by some solvers like `CONOPT,MINOS,LGO, IPOPT with GAMS`. Can it be solved with Yalmip?, If it can,  is it convenient to give an example, such as IPOPT solvers or other solvers?
Thank you very much!

**Comment by johanlofberg** [comment, marked as answer]
You would have to define what you are doing the max vs the min over, and what he nonlinearity actually is. It could be anything from trivial to intractable

If you want to use a pwa approximation, https://yalmip.github.io/command/interp1/

**Reply by BaiYansong-bys** [reply]
Thank you, Johan. **Min** refers to the minimum deviation between piecewise linear function and nonlinear function. **Max** refers to take the maximum value of linear functions. How can I define the **max** and **min** with Yalmip?  And further to  get  the parameter **a** and **b**?

**Reply by johanlofberg** [reply]
If you want to minimize the largest deviation between a linear approximation of a nonlinear function, you simply grid in x and solve the resulting LP

```
q = @(x)x.^2+.2*sin(5*x);
x = -0.5:0.05:1;
sdpvar a b
optimize([],norm(q(x)-(a*x+b),inf))
clf;plot(x,q(x));hold on
plot(x,value(a)*x+value(b))
```
